.github/workflows/docker.yaml

```diff
@@ -314,17 +314,24 @@ jobs:
       - name: Get canonical image name
         id: name
         run: echo "canonical=$(echo ${{ needs.image.outputs.images }} | sed 's/ .*//')" >> "$GITHUB_OUTPUT"
-      - name: Pull images
-        run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            docker pull "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done
       - name: Create and push manifest
         run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            echo "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done | \
-          xargs docker manifest create "${{ steps.name.outputs.canonical }}"
+          # Sometimes GCR gives random errors. Let's try a few times before giving up.
+          for RETRY in $(seq 10) ; do
+            for ARCH in ${{ inputs.archs }} ; do
+              if ! docker pull "${{ steps.name.outputs.canonical }}-${ARCH}" ; then
+                echo "Attempt #${RETRY} failed."
+                sleep 30
+                continue
```

Comment:
With these nested `for` loops, will this `continue` reference the correct one? In my head it will just move `ARCH` forward instead of `RETRY`.

Reply:
Argh! You're right. Fixed.